shuup/admin/modules/orders/views/detail.py

```diff
@@ -28,6 +28,12 @@ class OrderDetailView(DetailView):
     def get_toolbar(self):
         return OrderDetailToolbar(self.object)
 
+    def get_queryset(self):
+        qs = super(OrderDetailView, self).get_queryset().exclude(deleted=True)
+        if not self.request.user.is_superuser:
+            qs = qs.filter(shop=self.request.shop)
```

Comment:
never use `request.shop` in admin views, this is wrong. always use `get_shop(request)`

Reply:
Got it old codes here in diff. Carry on.